### PR TITLE
feat: improve summary

### PR DESCRIPTION
**Pull request by JorgePasco1**
## Summary by CodeRabbit

* **New Features**
  * Summaries include relative timestamps and direct links to original tweets
  * System prompt and summary formatting optimized for Telegram with configurable word limits

* **Changes**
  * Increased default summary processing from 50 to 100 tweets
  * New configurable limits: summary max tokens and summary max words

* **Tests**
  * Expanded test coverage across prompts, timestamps, links, delivery-failure scenarios, and edge cases

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/openai.rs [hunk 1]
```diff
@@ -27,31 +28,100 @@ struct Choice {
...
+
+    if hours < 0 || minutes < 0 {
+        // Future timestamp (shouldn't happen)
+        return "just now".to_string();
+    }
+
+    match hours {
+        0 => format!("{}m ago", minutes.max(1)),
+        1..=23 => format!("{}h ago", hours),
+        _ => tweet_time.format("%b %d, %H:%M").to_string(), // "Jan 15, 10:30"
```
[the remark this answers is not in the document]
Comment (JorgePasco1) [reply src/openai.rs]: Fixed
